File: go/hamming/hamming.go
package hamming

import (
	"errors"
)

// Distance calculates the hamming distance of two DNA strands.
func Distance(a, b string) (int, error) {

	aRunes, bRunes := []rune(a), []rune(b)

	if len(aRunes) != len(bRunes) {
		return 0, errors.New("strings not equal length")
	}

	var distance int

	for i := 0; i < len(aRunes); i++ {
		if aRunes[i] != bRunes[i] {
			distance++
		}
	}

	return distance, nil
}

File: php/hamming/hamming.php
<?php

function distance(string $strandA, string $strandB): int
{
    if (strlen($strandA) !== strlen($strandB)) {
        throw new InvalidArgumentException("DNA strands must be of equal length.");
    }

    return count(array_diff_assoc(str_split($strandA), str_split($strandB)));

}
